tensorizer/serialization.py

```diff
@@ -0,0 +1,1071 @@
+##############################################################################
+# serialization.py                                                   Wes Brown
+# Fast torch module/model serialization/deserialization     (c) 2023 Coreweave
+##############################################################################
+
+# try to import UNIX only dependencies
+try:
+    import fcntl
+    import resource
+except ImportError:
+    fcntl = None
+    resource = None
+
+from enum import Enum
+import collections.abc
+import io
+import os
+import tensorizer.stream_io as stream_io
+import tensorizer.utils as utils
+import mmap
+import numpy
+import struct
+import time
+import torch
+import typing
+import logging
+import tempfile
+import hashlib
+import zlib
+import ctypes
+
+if torch.cuda.is_available():
+    cudart = torch.cuda.cudart()
+else:
+    cudart = None
+
+from collections import OrderedDict
+from typing import Optional, Tuple, Union, List, Iterator, Dict, Callable, Any
+
+lz4 = None
+
+__all__ = ["TensorSerializer", "TensorDeserializer"]
+
+# Setup logger
+logger = logging.getLogger(__name__)
+
+
+# Whether the tensor is a parameter or a buffer on the model.
+class TensorType(Enum):
+    PARAM = 0
+    BUFFER = 1
+    STATE_DICT = 2
+
+
+TENSORIZER_VERSION = 1
+TENSORIZER_MAGIC = b"|TZR|"
+
+
+class TensorEntry(typing.TypedDict):
```

Comment:
`typing.TypedDict` is not supported until Python 3.8, but the project formally supports Python 3.6 and up. Python 3.6 is already deprecated, but Python 3.7 is not. We can either bump the officially supported version to 3.8 or change this type to a plain `dict`.

Reply:
I'd say bumping up to `3.8` is better than the options.